lib/ansible/plugins/filter/password_hash.yml

```diff
@@ -3,35 +3,35 @@ DOCUMENTATION:
   version_added: "historical"
   short_description: convert input password into password_hash
   description:
-    - Returns a password_hash of a secret
+    - Returns a password_hash of a secret.
   positional: _input
   notes:
-    - Algorithims available might be restricted by the system
+    - Algorithims available might be restricted by the system.
   options:
     _input:
-      description: secret to hash
+      description: Secret to hash.
       type: string
       required: true
     hashtype:
-      description: hashing algorithm to use
+      description: Hashing algorithm to use.
       type: string
       default: sha512
       choices: [ md5, blowfish, sha256, sha512 ]
     salt:
-      description: secret string that is used for the hashing, if none is provided a random one can be generated
+      description: Secret string that is used for the hashing, if none is provided a random one can be generated.
       type: int
     rounds:
-      description: number of encryption rounds, default varies by algorithim used
+      description: Number of encryption rounds, default varies by algorithim used.
       type: int
     ident:
-      description: no clue
-      type: string?
+      description: Algorithm identifier.
```

Comment:
The original text ('description: no clue`) made me laugh.

Reply:
Hehe, me too. I'm still not 100% sure `string` is correct (only 99.99%); I spent some time looking through the code of the libraries that are indirectly called here, and it looks like this is always a string, and if it's not it's converted to a string first, so `string` *should* be the correct value here.

(Actually finding *any* documentation on this is already really hard... I basically did not...)